File: 03-frontend-booking-system/booking-system-web/src/app/features/invitations/dtos/invitation-request-dto.ts
import { SlotIncludeMode } from "./slot-include-mode";

export class InvitationRequestDto {

    expiresAt?: string;
    maxUsage?: number;
    maxUsagePerIdentity?: number;
    requiredLogin!: boolean;
    slotIncludeMode!: SlotIncludeMode;
    slotIdList?: number[];
}

File: 02-backend-booking-system/src/main/java/com/littlek4za/booking_system/dtos/InvitationRequestDto.java
package com.littlek4za.booking_system.dtos;

import java.time.Instant;
import java.util.List;

import com.littlek4za.booking_system.validators.annotations.ValidInvitationRequest;
import com.littlek4za.booking_system.validators.annotations.ValidSlotIncludeMode;

import jakarta.validation.constraints.NotNull;

@ValidInvitationRequest
public record InvitationRequestDto(

    Instant expiresAt,
    
    Integer maxUsage,
    
    Integer maxUsagePerIdentity,

    @NotNull
    boolean requiredLogin,

    @ValidSlotIncludeMode
    String slotIncludeMode,

    List<Integer> slotIdList

) {

}
